### ISSUE TITLE
BIO_printf() doesn't handle "%I64u" format

### ISSUE THREAD

**Issue by gvanem** [openssl/openssl#670]
At several places in apps/*.c, `BIO_printf()` is called with formats like:
```
        BIO_printf(bio,
                   "---\nSSL handshake has read %"PRIu64" bytes and written %"PRIu64" bytes\n",
                   BIO_number_read(SSL_get_rbio(s)),
                   BIO_number_written(SSL_get_wbio(s)));
```
which for Windows at runtime gives:
```
SSL handshake has read 64u bytes and written 64u bytes
```

This is because `BIO_printf()` doesn't handle the `I64` modifier for Win32/64 targets.

=======================================
**Clarification**:

In `e_os.h`, there is:
```
# ifndef PRIu64
..
#   define PRIu64 "lu"
..
```
But this is too late; on TDM-gcc etc. `<inttypes.h>` which defines  `#define PRIu64 "I64u"`, has already been included. Thus the code expands to:
```
        BIO_printf(bio,
                   "---\nSSL handshake has read %I64u bytes and written %I64u bytes\n",
                   BIO_number_read(SSL_get_rbio(s)),
                   BIO_number_written(SSL_get_wbio(s)));
```
which `crypto/bio/b_print.c` doesn't understand.

The fix? 
* Use `"%llu"` and ignore the warning from gcc.
* Handle the `I64` modifier in `crypto/bio/b_print.c`.
* some other fix?

**Comment by gvanem** [comment]
BTW. my feeble attempt at patching `b_print.c` to support the `I64` modifier:
```
--- a/crypto/bio/b_print.c 2016-01-29 10:14:52
+++ b/crypto/bio/b_print.c 2016-02-13 12:52:34
@@ -124,6 +124,10 @@
 # define LLONG long
 #endif

+#if defined(_WIN32) && !defined(__CYGWIN__)
+#  define HAVE_I64_FMT
+#endif
+
 static void fmtstr(char **, char **, size_t *, size_t *,
                    const char *, int, int, int);
 static void fmtint(char **, char **, size_t *, size_t *,
@@ -278,6 +282,15 @@
                 cflags = DP_C_LDOUBLE;
                 ch = *format++;
                 break;
+#ifdef HAVE_I64_FMT
+            case 'I':
+                if (format[0] == '6' && format[1] == '4') {
+                    cflags = DP_C_LLONG;
+                    format += 2;
+                    ch = *format++;
+                }
+                break;
+#endif
             default:
                 break;
             }
```
Works fine in a `openssl s_client -connect www.ssllabs.com:443` test:
```
SSL handshake has read 4414 bytes and written 406 bytes
```

**Comment by gvanem** [comment]
I'm not sure why the *problem* vanished since I filed this issue. But it seems the pre-processed output of this (`apps/s_client.c`): 
```
 BIO_printf(bio,
              "---\nSSL handshake has read %"PRIu64" bytes and written %"PRIu64" bytes\n",
   ...
```
now expands into:
```
    BIO_printf (bio,
                "---\nSSL handshake has read %" "llu" " bytes and written %" "llu" " bytes\n",
   ...
```

(thus correctly handled in `crypto/bio/b_print.c`). I've only checked `apps/s_client.c` using MSVC-2015. I've not checked using TDM-gcc.

So I think the issue could be closed unless some MingW/TDM-gcc user can confirm there is still an issue with the use of `PRIu64` in OpenSSL.
